File: src/main/java/com/example/dq/model/ChunkRecord.java
package com.example.dq.model;

/** scan_chunk 记录 */
public record ChunkRecord(
        long id,
        long scanTableId,
        int seq,
        String rangeStart,
        String rangeEnd,
        boolean nullChunk,
        ScanStatus status,
        long rowCount,
        String colStatsJson,
        int attempts
) {

    public Range toRange() {
        return new Range(rangeStart, rangeEnd, nullChunk);
    }
}

File: common/src/main/kotlin/com/example/dq/model/ChunkRecord.kt
package com.example.dq.model

/** scan_chunk 记录 */
data class ChunkRecord(
    val id: Long,
    val scanTableId: Long,
    val seq: Int,
    val rangeStart: String?,
    val rangeEnd: String?,
    val nullChunk: Boolean,
    val status: ScanStatus?,
    val rowCount: Long,
    val colStatsJson: String?,
    val attempts: Int
) {

    fun toRange(): Range {
        return Range(rangeStart, rangeEnd, nullChunk)
    }
}
